src/libcephsqlite.cc

```diff
@@ -1,473 +1,474 @@
+  ceph_assert(lock == SQLITE_LOCK_NONE || (lock > SQLITE_LOCK_NONE && f->io.rs.is_locked()));
+  ceph_assert(lock >= eLock);
+  if (eLock <= SQLITE_LOCK_NONE && SQLITE_LOCK_NONE < lock) {
+    if (int rc = f->io.rs.unlock(); rc < 0) {
+      df(5) << "failed: " << rc << dendl;
+      return SQLITE_IOERR;
+    }
+  }
 
+  lock = eLock;
   return SQLITE_OK;
 }
 
-/*
-** Truncate a file. This is a no-op for this VFS (see header comments at
-** the top of the file).
-*/
-static
-int _cephsqlite3_Truncate(sqlite3_file *pFile, sqlite_int64 size)
+static int CheckReservedLock(sqlite3_file *pFile, int *pResOut)
 {
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << std::endl;
-  CephFile *p = (CephFile*)pFile;
-  libradosstriper::RadosStriper *rs = get_radosstriper(get_db_name(p->name).c_str());
+  auto f = (cephsqlite_file*)pFile;
+  df(5) << dendl;
 
-  if (rs->trunc(p->name, size) != 0)
-    return SQLITE_IOERR;
+  auto& lock = f->lock;
+  if (lock > SQLITE_LOCK_SHARED) {
+    *pResOut = 1;
+  }
 
+  df(10);
+  f->io.rs.printlockers(*_dout);
+  *_dout << dendl;
+
+  *pResOut = 0;
   return SQLITE_OK;
 }
 
-/*
-** Sync the contents of the file to the persistent media.
-*/
-static
-int _cephsqlite3_Sync(sqlite3_file *pFile, int flags)
+static int Close(sqlite3_file *pFile)
 {
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << std::endl;
+  auto f = (cephsqlite_file*)pFile;
+  df(5) << dendl;
+  f->~cephsqlite_file();
   return SQLITE_OK;
 }
 
-
-/*
-** Write the size of the file in bytes to *pSize.
-*/
-static
-int _cephsqlite3_FileSize(sqlite3_file *pFile, sqlite_int64 *pSize)
+static int Read(sqlite3_file *pFile, void *zBuf, int iAmt, sqlite_int64 iOfst)
 {
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << std::endl;
-  CephFile *p = (CephFile*)pFile;
-  libradosstriper::RadosStriper *rs = get_radosstriper(get_db_name(p->name).c_str());
+  auto f = (cephsqlite_file*)pFile;
+  df(5) << zBuf << " " << iOfst << "~" << iAmt << dendl;
 
-  uint64_t  size = 0;
-  time_t    mtime = 0;
-  int       rc = rs->stat(p->name, &size, &mtime);
+  bufferlist bl;
+  bufferptr bp = ceph::buffer::create_static(iAmt, (char*)zBuf);
+  bl.push_back(std::move(bp));
+  if (int rc = f->io.rs.read(zBuf, iAmt, iOfst); rc < 0) {
+    df(5) << "read failed: " << cpp_strerror(rc) << dendl;
+    return SQLITE_IOERR_READ;
+  } else {
+    df(5) << "= " << rc << dendl;
+    if (rc < iAmt) {
+      memset(zBuf, 0, iAmt-rc);
+      return SQLITE_IOERR_SHORT_READ;
+    } else {
+      return SQLITE_OK;
+    }
+  }
+}
 
-  *pSize = 0;
+static int Write(sqlite3_file *pFile, const void *zBuf, int iAmt, sqlite_int64 iOfst)
+{
+  auto f = (cephsqlite_file*)pFile;
+  df(5) << iOfst << "~" << iAmt << dendl;
 
-  if (rc == 0)
-    *pSize = (sqlite_int64)size;
+  auto aiocp = aiocompletionptr(librados::Rados::aio_create_completion());
+  ceph::bufferlist bl;
+  bl.append((const char*)zBuf, iAmt);
```

Comment:
`aiocp` and `bufferlist bl;` don't seem to be used anywhere in this function ?

Reply:
Ah, this is stale code that needs removed. Thanks!